flake.nix

```diff
@@ -21,82 +21,85 @@
+                    algebraic-graphs = with hf;
+                      dontCheck
                         (callCabal2nix "algebraic-graphs" (inputs.alga) { });
 
-                      dependent-hashmap = with hf;
-                        unmarkBroken (dontCheck hp.dependent-hashmap);
+                    dependent-hashmap = with hf;
+                      unmarkBroken (dontCheck hp.dependent-hashmap);
 
-                      diagnose = hf.callCabal2nixWithOptions "diagnose" (inputs.diagnose) "-fmegaparsec-compat" {};
+                    diagnose = hf.callCabal2nixWithOptions "diagnose" (inputs.diagnose) "-fmegaparsec-compat" { };
 
-                      eclair-lang = with hf;
-                        (callCabal2nix "eclair-lang" ./. { }).overrideAttrs
+                    eclair-lang = with hf;
+                      (callCabal2nix "eclair-lang" ./. { }).overrideAttrs
                         (o: {
                           version = "${o.version}.${version}";
                           checkPhase = ''
-                          runHook preCheck
-                          DATALOG_DIR="${o.src}/cbits/" SOUFFLE_BIN="${pkgs.souffle}/bin/souffle" ./Setup test
-                          runHook postCheck
+                            runHook preCheck
+                            DATALOG_DIR="${o.src}/cbits/" SOUFFLE_BIN="${pkgs.souffle}/bin/souffle" ./Setup test
+                            runHook postCheck
                           '';
                         });
-                    };
-                };
-            in { inherit haskellPackages; };
-          pkgs = import np {
-            inherit system config;
-            overlays = [
-              ds.overlay
-              shs.overlay."${system}"
-              llvm-cg.overlay."${system}"
-              overlay
+                  };
+              };
+          in
+          { inherit haskellPackages; };
+        pkgs = import np {
+          inherit system config;
+          overlays = [
+            ds.overlay
+            shs.overlay."${system}"
+            llvm-cg.overlay."${system}"
+            overlay
+          ];
+        };
+      in
+      with pkgs.lib; rec {
+        inherit overlay;
+        packages = { inherit (pkgs.haskellPackages) eclair-lang; };
+        defaultPackage = packages.eclair-lang;
+        devShell = pkgs.devshell.mkShell {
+          name = "ECLAIR-LANG";
+          imports = [ (pkgs.devshell.importTOML ./devshell.toml) ];
+          packages = with pkgs;
+            with haskellPackages; [
+              souffle
+              pkgs.llvmPackages_14.llvm.dev
+              pkgs.ghcid
+              (ghcWithPackages (p:
+                with p; [
+                  algebraic-graphs
+                  hspec-discover
+                  llvm-codegen
+                  souffle-haskell
+                  ghc
+                  cabal-install
+                  hsc2hs
+                  hpack
+                  haskell-language-server
+                  hlint
```

Comment:
Just making sure: `hlint` is the only thing you added here? Hard to see with the diff on github.. :sweat_smile:

Reply:
ouch, I've commited too quickly